File: TOMIMSV4/Entity/Interface/Pedido_Compra/Pedido_Compra_Det/clsBeI_nav_ped_compra_det.vb
Public Class clsBeI_nav_ped_compra_det
    Implements ICloneable
    Public Property NoEnc() As String = ""
    Public Property Line_No() As Integer = 0
    Public Property Variant_Code() As String = ""
    Public Property No() As String = ""
    Public Property Type() As New Object
    Public Property Description() As String = ""
    Public Property Description2() As String = ""
    Public Property Location_Code() As String = ""
    Public Property Quantity() As Decimal = 0.0
    Public Property Unit_of_Measure_Code() As String = ""
    Public Property Direct_Unit_Cost() As Decimal = 0.0
    Public Property Line_Amount() As Decimal = 0.0
    Public Property Quantity_Received() As Decimal = 0.0
    Public Property Planed_Receipt_Date() As Date = Now.Date
    Public Property Barcode() As String = ""
    Public Property Size() As String = ""
    Public Property Color() As String = ""
    Sub New()
    End Sub

    Sub New(ByRef NoEnc As String, ByVal No As String, ByVal Type As String, ByVal Description As String, ByVal Description2 As String, ByVal Location_Code As String, ByVal Quantity As Decimal, ByVal Unit_Of_Measure_Code As String, ByVal Direct_Unit_Cost As Decimal, ByVal Line_Amount As Decimal, ByVal Quantity_Received As Decimal, ByVal Planed_Receipt_Date As String)
        Me.NoEnc = NoEnc
        Me.No = No
        Me.Type = Type
        Me.Description = Description
        Me.Description2 = Description2
        Me.Location_Code = Location_Code
        Me.Quantity = Quantity
        Me.Unit_of_Measure_Code = Unit_Of_Measure_Code
        Me.Direct_Unit_Cost = Direct_Unit_Cost
        Me.Line_Amount = Line_Amount
        Me.Quantity_Received = Quantity_Received
        Me.Planed_Receipt_Date = Planed_Receipt_Date
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/I_nav_Ped_Compra/clsBeI_nav_ped_compra_det.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMSWebAPI.Be
{
    public class clsBeI_nav_ped_compra_det : ICloneable
    {
        [Column("NoEnc")]
        [DisplayName("NoEnc")]
        public string NoEnc { get; set; } = "";

        [Column("No")]
        [DisplayName("No")]
        public string No { get; set; } = "";

        [Column("Line_No")]
        [DisplayName("Line_No")]
        public int Line_No { get; set; } = 0;

        [Column("Type")]
        [DisplayName("Type")]
        public string Type { get; set; } = "";

        [Column("Description")]
        [DisplayName("Description")]
        public string Description { get; set; } = "";

        [Column("Description2")]
        [DisplayName("Description2")]
        public string Description2 { get; set; } = "";

        [Column("Location_Code")]
        [DisplayName("Location_Code")]
        public string Location_Code { get; set; } = "";

        [Column("Quantity")]
        [DisplayName("Quantity")]
        public double Quantity { get; set; } = 0;

        [Column("Unit_Of_Measure_Code")]
        [DisplayName("Unit_Of_Measure_Code")]
        public string Unit_Of_Measure_Code { get; set; } = "";

        [Column("Direct_Unit_Cost")]
        [DisplayName("Direct_Unit_Cost")]
        public double Direct_Unit_Cost { get; set; } = 0;

        [Column("Line_Amount")]
        [DisplayName("Line_Amount")]
        public double Line_Amount { get; set; } = 0;

        [Column("Quantity_Received")]
        [DisplayName("Quantity_Received")]
        public double Quantity_Received { get; set; } = 0;

        [Column("Planed_Receipt_Date")]
        [DisplayName("Planed_Receipt_Date")]
        public DateTime Planed_Receipt_Date { get; set; } = DateTime.Now;

        [Column("Variant_Code")]
        [DisplayName("Variant_Code")]
        public string Variant_Code { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        public string Barcode { get; set; } = "";
        public string Size { get; set; } = "";
        public string Color { get; set; } = "";
        public  int LayersPallet { get; set; } = 0;
        public int BoxesLayer { get; set; } = 0;    
        public clsBeI_nav_ped_compra_det() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
